src/cargo/util/lints.rs

```diff
@@ -875,3 +875,45 @@ pub fn unused_dependencies(
     }
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+
+    #[test]
+    fn ensure_sorted_lints() {
+        // This will be printed out if the fields are not sorted.
+        let location = std::panic::Location::caller();
+        println!("\nTo fix this test, sort `LINTS` in {}\n", location.file(),);
+
+        let actual = super::LINTS
+            .iter()
+            .map(|l| l.name.to_uppercase())
+            .collect::<Vec<_>>();
+
+        let mut expected = actual.clone();
+        expected.sort();
+        let expected = format!("{:#?}", expected);
+        let actual = format!("{:#?}", actual);
+        snapbox::assert_data_eq!(actual, expected);
```

Comment:
nit: `snapbox` has a `ToDebug::to_debug()` so this can be `snapbox::assert_data_eq!(actual.to_debug(), expected.to_debug())`